inference-engine/src/mkldnn_plugin/mkldnn_node.cpp

```diff
@@ -839,14 +839,27 @@ void MKLDNNNode::prepareMemory(const NodeDesc *selected_pd, mkldnn::primitive_de
     }
 }
 
-void MKLDNNNode::initInPlace() {
-    auto selected_pd = getSelectedPrimitiveDescriptor();
-    if (selected_pd == nullptr)
-        IE_THROW() << "Preferable primitive descriptor is not set.";
-    auto config = selected_pd->getConfig();
-
-    for (auto &in : config.inConfs) if (in.inPlace >= 0) inplace = true;
-    for (auto &out : config.outConfs) if (out.inPlace >= 0) inplace = true;
+bool MKLDNNNode::isInPlace() {
+    if (inplace == InPlaceType::Unknown) {
+        auto selected_pd = getSelectedPrimitiveDescriptor();
+        if (selected_pd == nullptr)
+            IE_THROW() << "Preferable primitive descriptor is not set.";
+        inplace = InPlaceType::NoInPlace;
+        auto config = selected_pd->getConfig();
+        for (auto &in : config.inConfs) {
+            if (in.inPlace >= 0) {
+                inplace = InPlaceType::InPlace;
+                break;
+            }
+        }
+        for (auto &out : config.outConfs) {
+            if (out.inPlace >= 0) {
+                inplace = InPlaceType::InPlace;
```

Comment:
This parameter should be updated every type we change `selectedPrimitiveDescriptorIndex` or better reset to unknown, to avoid possible errors.

Reply:
Yes, you're right. Thank you!

I returned method `InitInPlace`. It should be called every time `selectedPrimitiveDescriptorIndex` is changed to avoid some problems.